MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c

```diff
@@ -1947,6 +1947,109 @@ InstallAcpiTableFromHob (
   return Status;
 }
 
+/**
+  This function will get ACPI table from gAcpiTableHobGuid Guid hob and install Acpi table from it.
+
+  @param  AcpiTableInstance  Protocol instance private data.
+  @param  GuidHob            GUID HOB header.
+
+  @return EFI_SUCCESS        The function completed successfully.
+  @return EFI_NOT_FOUND      The function doesn't find the Rsdp from AcpiSiliconHob.
+  @return EFI_ABORTED        The function could not complete successfully.
+
+**/
+EFI_STATUS
+InstallAcpiTableFromAcpiSiliconHob (
+  EFI_ACPI_TABLE_INSTANCE   *AcpiTableInstance,
+  EFI_HOB_GUID_TYPE         *GuidHob
+)
+{
+  ACPI_SILICON_HOB                                 *AcpiSiliconHob;
+  EFI_ACPI_3_0_ROOT_SYSTEM_DESCRIPTION_POINTER     *SiAcpiHobRsdp;
+  EFI_ACPI_DESCRIPTION_HEADER                      *SiCommonAcpiTable;
+  EFI_STATUS                                       Status;
+  UINTN                                            EntrySize;
+  UINT8                                            *TempBuffer;
+  EFI_ACPI_TABLE_VERSION                           Version;
+  UINTN                                            NumOfTblEntries;
+
+
+  DEBUG ((DEBUG_INFO, "InstallAcpiTableFromAcpiSiliconHob\n"));
+  //
+  // Initial variable.
+  //
+  SiAcpiHobRsdp      = NULL;
+  SiCommonAcpiTable  = NULL;
+  AcpiSiliconHob     = GET_GUID_HOB_DATA (GuidHob);
+  Status             = EFI_SUCCESS;
+  Version            = PcdGet32 (PcdAcpiExposedTableVersions);
+  //
+  // Got RSDP table from ACPI Silicon Hob.
+  //
+  SiAcpiHobRsdp = (EFI_ACPI_3_0_ROOT_SYSTEM_DESCRIPTION_POINTER *) (AcpiSiliconHob->Rsdp);
+  if (SiAcpiHobRsdp == NULL) {
+    DEBUG ((DEBUG_ERROR, "InstallAcpiTableFromAcpiSiliconHob: Fail to locate RSDP Acpi table!!\n"));
+    return EFI_NOT_FOUND;
+  }
+
+  DEBUG ((DEBUG_INFO, "Silicon ACPI RSDP address : 0x%lx\n", SiAcpiHobRsdp));
+  AcpiTableInstance->Rsdp3 = SiAcpiHobRsdp;
+
+  if (SiAcpiHobRsdp->RsdtAddress != 0x00000000) {
+    //
+    // Initial RSDT.
+    //
+    TempBuffer = (UINT8 *) (UINTN) (SiAcpiHobRsdp->RsdtAddress);
+    SiCommonAcpiTable = (EFI_ACPI_DESCRIPTION_HEADER *) TempBuffer;
+    AcpiTableInstance->Rsdt3 = SiCommonAcpiTable;
+
+    if (SiCommonAcpiTable->Length <= sizeof (EFI_ACPI_DESCRIPTION_HEADER)) {
+      DEBUG ((DEBUG_ERROR, "RSDT length is incorrect\n"));
+      return EFI_ABORTED;
+    }
+    //
+    // Calcaue 32bit Acpi table number.
+    //
+    NumOfTblEntries = (SiCommonAcpiTable->Length - sizeof (EFI_ACPI_DESCRIPTION_HEADER)) / sizeof (UINT32);
+    AcpiTableInstance->NumberOfTableEntries1 = NumOfTblEntries;

```

Comment:
The FSP ACPI table is not relocated to an ACPI-reclaimed memory, instead, it is still in the FSP reserved memory range.
Is there an FSP spec change that reports the FSP ACPI table in correct (ACPI-reclaimed) memory type?

Reply:
As I know, FSP ACPI table PPI service is allocated EfiACPIReclaimMemory type memory for ACPI table under FSP phase, so ACPITableDxe can directly use ACPI table which from HOB.